### ISSUE TITLE
Convert Multiband Raster to WPS binary

### ISSUE THREAD

**Issue by taoansf** [GIS4WRF/gis4wrf#145]
Hello,

I'm trying do convert a multiband raster to WPS binary by GIS4WRF process tool
![image](https://user-images.githubusercontent.com/22460776/62050110-0ae74080-b1e7-11e9-9a27-eac67c061d62.png)

However the resulting output is a single band binary file.

I attached my input and and output data for your checking.
[green-mask-final-small.zip](https://github.com/GIS4WRF/gis4wrf/files/3442368/green-mask-final-small.zip)
[greenfrac_fpar_modis-07-2019-wps.zip](https://github.com/GIS4WRF/gis4wrf/files/3442317/greenfrac_fpar_modis-07-2019-wps.zip)

Any help will be very appreciated.

With best regards,

Taoan

**Comment by dmey** [comment]
I do not quite understand why you want to convert a multi band to WPS binary but in any case this feature is not supported. You need to convert to single band first.

**Comment by taoansf** [comment]
My idea is to update geogrid database to increase coast line resolution in my WRF simulations.
I got good improvements with Landuse data, which is single band. However, Albedo, Greenfraction and others multiband data I can't do because it is not working with GIS4WRF conversion tool.

If you know another way to do this, please share it with me.

Thank you for your attention.

Taoan

**Comment by dmey** [comment]
We do not support that feature at the moment. @letmaik I was going to add this as a feature request under the `future` milestone -- I am not sure we should aim to add this at the next release due to time...

**Comment by letmaik** [comment]
That's right, this is [not supported currently](https://github.com/GIS4WRF/gis4wrf/blob/f32c38ff9c1e7bb9ad822a020d6b3852755af180/gis4wrf/core/writers/wps_binary.py#L123) and there is no work-around for this either. We should really print an error message in that case instead of simply converting the first band.

@dmey `future` sounds fine, it's not a simple fix but requires some work and I don't think we have the resources currently to do this in the near future.

@taoansf Contributions are always welcome. The code is here: https://github.com/GIS4WRF/gis4wrf/blob/master/gis4wrf/core/writers/wps_binary.py.

**Comment by taoansf** [comment]
Thanks all
I will study wps_binary.py to understand better the process.

I also found this explanation about wps format that give some insights http://www.met.wur.nl/Summerinthecity/posts/2014/03/wrfgisinput/

 If wps format is similar to ENVI image format, so I will look for some solution by this way using GDAL. 

However, I have another question related with "wordsize" parameter on wps index file. I got this error when I try to convert a geotiff float32 to wps and open the output with GIS4WRF.
![image](https://user-images.githubusercontent.com/22460776/62076147-4a308400-b21d-11e9-9f9e-8f366737fed8.png)

Is there some trick for conversion of geotiff and get a "wordsize=2"? because I only get wordsize=4 on my output.

These are my test files.
[SOILTEMP.zip](https://github.com/GIS4WRF/gis4wrf/files/3443987/SOILTEMP.zip)

**Comment by letmaik** [comment]
Currently wordsize is determined automatically with a heuristic with the goal of trying to represent the input as accurately as possible. Feel free to open a separate feature request about this as it should really be an option for the user to choose it manually if needed, typically to reduce data volume.

The error you got is correct and a limitation of GDAL which doesn't support this high bit depth natively. Again, feel free to open a feature request for that. The way to do it would be to preprocess the WPS data (downsampling) before handing it to GDAL.

**Comment by taoansf** [comment]
Hello samp7,

I'm sorry for the late answer. 
My sugestion is to adopt the WGS84 datum epsg:4326 to avoid conflicts.
This article can give you some orientations https://fabienmaussion.info/2018/01/06/wrf-projection/
I also recomend this plugin for raster edition https://github.com/SMByC/ThRasE

I hope this help you.

With best regards,

Taoan
